### DISCUSSION TITLE
values not being reactive when accessing from a custom library

### DISCUSSION THREAD

**Discussion by kaueDM** [pmndrs/zustand#1420] [Q&A]
I created an UI library which uses zustand to control the active theme (`dark` or `light`). Something like:

```ts
type Store = {
  mode: 'light' | 'dark'
  toggleMode: () => void
}

export default create<Store>(
  persist(
    (set) => {
      return {
        mode: Appearance.getColorScheme() || 'dark',
        toggleMode: () => {
          return set((store: Store) => ({
            mode: store?.mode === 'light' ? 'dark' : 'light'
          }))
        }
      }
    },
    {
      name: 'ReactNativeComponentsStore',
      getStorage: () => AsyncStorage
    }
  )
)
```

There are tons of components and a "provider" component which receives some parameters (not relevant to the issue). This provider wraps the entire app, so the internal zustand store should be accessible from any part of the app. This works fine and I can access zustand values using a function exported from the same UI library

```ts
// library
const useMode = () => {
  const values = store((state) => {
    return {
      mode: state?.mode,
      toggleMode: state?.toggleMode
    }
  })

  return values
}

// somewhere inside the app
import { useMode } from 'mylib' 

  const { mode, toggleMode } = useMode()

  return (
    <View>
      <Text>{mode}</Text>

      <Button onPress={toggleMode} />
    </View>
  )
```

It's possible to view the initial value of `mode` and toggle it (It's actually toggling, components imported from the lib which have different styles based on `mode` update as expected and the lib logs the updated value), but `mode` inside the app never updates, only reopening it.

I tried using a vanilla store (I have another library with a similar use-case which runs a vanilla store), but I got the same result.

**Comment by dai-shi** [comment]
Can you create as minimal reproduction as possible to explain the problem (without any other libs) with codesandbox http://react.new ?

**Reply by kaueDM** [reply]
@dai-shi here is the problem:

![image](https://user-images.githubusercontent.com/20171863/203569220-6b44615c-e4f4-470a-a03a-d348d7185775.png)

There are two different instances of the same store, which don't make much sense to me. How is this even possible? I'll try to spin up a repro.
